assets/src/edit-story/app/media/categories/reducer.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import * as types from './types';
+
+export const INITIAL_STATE = {
```

Comment:
Should we have the same for media? So a media sub-state that holds the loading and things like the token etc.?

Reply:
I assumed that categories would be at the same level as where "searchTerm" is stored. So you'd just add a "categories" array to media3p/reducer.js?

Reply:
I thought about that. But then there are also 4 loading booleans in one level which is kind of aweful. Wdyt?

Reply:
I don't think that's a good enough reason. If we want to introduce a level of indirection there (wrap the loading booleans in an object) that's fine, but that shouldn't dictate where the state lives.

Reply:
Having a categories list for each provider makes sense to me, as each provider returns their own category list. That way when switching between providers, we don't need to clear the media3p/categories state and wait for it to load again.

Reply:
So are we fine with keeping it like that? Making the root state bigger does not improve it I guess.

Reply:
Ah fair enough, categories per provider makes sense. This LGTM.